pkg/cmd/server/apis/config/v1/types.go

```diff
@@ -770,6 +774,20 @@ type OAuthTemplates struct {
 	Error string `json:"error"`
 }
 
+// ExternalOAuthConfig allos to configure external OAuth server discovery
+type ExternalOAuthConfig struct {
+	// MetadataFile is a path to a file containing the discovery endpoint for OAuth 2.0 Authorization Server Metadata
+	// for an External OAuth server.
+	// See IETF Draft: // https://tools.ietf.org/html/draft-ietf-oauth-discovery-04#section-2
+	MetadataFile string `json:"metadataFile"`
+
+	// MasterPublicURL is used for building valid client redirect URLs for internal and external access
+	MasterPublicURL string `json:"masterPublicURL"`
+
+	// AssetPublicURL is used for building valid client redirect URLs for external access
+	AssetPublicURL string `json:"assetPublicURL"`
```

Comment:
This comes from the `openshift-web-console` configmap called `webconsole-config` under the `webconsole-config.yaml` key in this spot: https://github.com/openshift/api/blob/master/webconsole/v1/types.go#L31 .

I'm in full support of publishing a `public-config` configmap with a `publicURL` key in the `openshift-web-console` namespace if you'd prefer to avoid digging that far into the existing configmap.

Reply:
Nothing seems to be using ConcolePublicURL and the comment on it is doubtful we should even have that field. So I do nto see how I can swap the current use of the homonymous field on OAuthConfig with that field in the config maps ? Can I even assume it is properly populated ?

Reply:
Given usage, I could probably get behind a `WebConsoleProxyConfig` stanza if need be, but I'm not seeing why an external oauth provider wouldn't just set this to their fancy